app/controllers/payment-links/get-edit-controller.js

```diff
@@ -17,7 +17,9 @@ module.exports = (req, res) => {
     self: formattedPathFor(paths.paymentLinks.edit, req.params.productExternalId),
     editInformation: formattedPathFor(paths.paymentLinks.editInformation, req.params.productExternalId),
     editReference: formattedPathFor(paths.paymentLinks.editReference, req.params.productExternalId),
-    editAmount: formattedPathFor(paths.paymentLinks.editAmount, req.params.productExternalId)
+    editAmount: formattedPathFor(paths.paymentLinks.editAmount, req.params.productExternalId),
+    addMetadata: formattedPathFor(paths.paymentLinks.metadata.add, req.params.productExternalId),
+    editMetadata: formattedPathFor(paths.paymentLinks.metadata.edit, req.params.productExternalId)
```

Comment:
Could `formattedPathFor` be used to also calculate the metatada key specific URL? that would avoid doing any mutation in the template. 

Two options for this might be: 
1. pass in `formattedPathFor` into the template -- this could then look like
```html
<a class="govuk-link" href="{{ formattedPathFor(paths.paymentLinks.metadata.edit, product.external_id, metadataKey) }}">
``` 

2. An easy to unit test method to add the edit path to the metadata itself before passing it into the template
```
product.metadata.forEach((metadata) => metadata.editUrl = formattedPathFor(...))
```
```
<a class="govuk-link" href="{{ metadata.editUrl }}">
``` 

The current option looks fine to me as well, happy to let the template do this just a little harder to test.

Reply:
The latter option is tricky because the metadata is just an object (e.g `{"key1": "value1", "key2": "value2"}`) so there’s nowhere to really put the edit URL. We could convert it to something else (an array of objects, say) but that would be a bit of a pain, so I went with the first option, even though it’s not as great.